mypyc/analysis/dataflow.py

```diff
@@ -308,6 +319,11 @@ def visit_assign(self, op: Assign) -> GenAndKill:
             return set(), {op.dest}
         return set(), set()
 
+    def visit_set_mem(self, op: SetMem) -> GenAndKill:
+        if op.dest in self.args:
+            return set(), {op.dest}
+        return set(), set()
```

Comment:
This should be similar to `visit_register_op`, since we don't modify `op.dest`, but the value pointed to by it.